rasa/shared/core/training_data/story_writer/yaml_story_writer.py

```diff
@@ -228,7 +228,26 @@ def process_user_utterance(
             entities = []
             for entity in user_utterance.entities:
                 if "value" in entity:
-                    entities.append(OrderedDict([(entity["entity"], entity["value"])]))
+                    if user_utterance.__dict__.get("predicted_entities"):
```

Comment:
How about having a new method on `WronglyClassifiedUserUtterance` like `inline_comment_for_entitiy(entity)` - then you could just check if the method exists like this line `hasattr(user_utterance, "inline_comment")`? This would also mean keeping the functionality encapsulated in `WronglyClassifiedUserUtterance`.

The issue with moving `WronglyClassifiedUserUtterance` to shared is that it can then be used by Rasa X and we would have to be careful about ever updating it.